src/main/java/az/ingress/mscategory/service/CategoryService.java

```diff
@@ -0,0 +1,47 @@
+package az.ingress.mscategory.service;
+
+import az.ingress.mscategory.annotation.Log;
+import az.ingress.mscategory.dao.entity.CategoryEntity;
+import az.ingress.mscategory.dao.repository.CategoryRepository;
+import az.ingress.mscategory.model.response.CategoryDto;
+import lombok.RequiredArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.stereotype.Service;
+
+import java.util.*;
+
+@Slf4j
+@Service
+@RequiredArgsConstructor
+public class CategoryService {
+
+    private final CategoryRepository categoryRepository;
+
+    @Log
+    public List<CategoryDto> getCategoriesWithSubCategories() {
+        List<CategoryEntity> categoryEntities = categoryRepository.findAll();
```

Comment:
don't use findAll , try to find first in cache(redis) after if cache is empty or unavailable go to the DB